cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/service/DistributedStreamService.java

```diff
@@ -258,8 +259,13 @@ public void thresholdChanged(String streamName, int threshold) {
       getStreamCoordinatorClient().addListener(streamName, new StreamPropertyListener() {
         @Override
         public void generationChanged(String streamName, int generation) {
-          Cancellable previousAggregator = aggregators.replace(streamName, createSizeAggregator(streamName, 0,
-                                                                                                threshold));
+          int currentThresold = threshold;
+          StreamSizeAggregator currentAggregator = aggregators.get(streamName);
+          if (currentAggregator != null) {
+            currentThresold = currentAggregator.getStreamThresholdMB();
```

Comment:
hum. Instead of creating a new aggregator, is it sufficient to just have a `reset()` method on the aggregator to reset the size to 0?